src/pages/blog/tag/[tag]/[...page].astro

```diff
@@ -0,0 +1,99 @@
+---
+import BaseLayout from "@layouts/BaseLayout.astro";
+import HorizontalCard from "@components/HorizontalCard.astro";
+import { getCollection } from "astro:content";
+
+export async function getStaticPaths({ paginate }) {
+    const all_posts = await getCollection("blog");
+    const all_tags = all_posts.flatMap((post) => {
+        return post.data.tags || [];
+    });
+
+    return all_tags.flatMap((tag) => {
+        const filtred_posts = all_posts.filter((post) => {
+            return post.data.tags?.includes(tag);
+        });
+        return paginate(filtred_posts, {
+            params: { tag },
+            pageSize: 10,
+        });
+    });
+}
+
+const { page } = Astro.props;
+const params = Astro.params;
+---
+
+<BaseLayout title="Blog" sideBarActiveItemID="blog">
+    <div class="mb-5">
+        <div class="text-3xl w-full font-bold">Blog</div>
+    </div>
+    {
+        page.data.length === 0 ? (
+            <div class="bg-base-200 border-l-4 border-secondary w-full p-4 min-w-full">
+                <p class="font-bold">Sorry!</p>
+                <p>
+                    There are no blog posts to show at the moment. Check back
+                    later!
+                </p>
+            </div>
+        ) : (
+            <ul>
+                {page.data.map((post) => (
+                    <>
+                        <HorizontalCard
```

Comment:
As we are going to add support for tags, we should include the tags in all blog post cards. Let's add the `tags={post.data.tags}` here